08_traits_and_reading_bytes.md

```diff
@@ -1,20 +1,38 @@
 # Traits and Reading Bytes
 
-One way you might consider reading the rest of the data from the transaction is to use various ranges. For example, consider the following code:
+One way you might consider reading the rest of the data from the transaction is to use various ranges.
+For example, consider the following code:
 
 ```rust
 let transaction_bytes = hex::decode(transaction_hex).unwrap();
 let version = u32::from_le_bytes(&transaction_bytes[0..4]);
 let number_of_inputs = u32::from_le_bytes(&transaction_bytes[5..6]);
 ```
 
-Notice how we're grabbing different ranges of `transaction_bytes`. We have to repeatedly reference `transaction_bytes` and we have to keep track of the start and end indexes for each component. This is not ideal. Transactions are presented in hex format for a reason. They are designed to be serialized as byte streams that can be transmitted over the network and read one byte at a time in order.
-
-One way to read a byte stream is to leverage Rust's standard library's [`Read`](https://doc.rust-lang.org/std/io/trait.Read.html) trait. The slice data type in Rust implements the `Read` trait. What does this mean? Well, as we will see, it gives us a method, `read`, which will read some bytes from the slice and then store that data into a array. When we call `read` again, it will start from where it left off. In other words, it keeps track of where we are in the stream and modifies the pointer as it reads. This means we don't need to keep track of any indexes.
+Notice how we're grabbing different ranges of `transaction_bytes`.
+We have to repeatedly reference `transaction_bytes` and we have to keep track of the start and end indexes for each component.
+This is not ideal because we can easily make mistakes.
+
+*Note: there's an indexing mistake in the code above, can you see what it is?*
+<!--
+Since vec[0..4] notation is not inclusive to the end of the index, we are skipping one byte: the element at index 4.
+-->
+
+Transactions are presented in hex format for a reason.
+They are designed to be serialized as byte streams that can be transmitted over the network and read one byte at a time in order.
+A better solution would be to automatically manage the indices and have a function with which we can just ask for the next bytes we require.
```

Comment:
Might read a little easier like so:
"A better solution would be to use a function that keeps track of the indices and allows us to request the number of bytes we require"